File: examples/Java CRUD demo/src/main/java/bryntum/scheduler/request/handler/ResourceSyncHandler.java
package bryntum.scheduler.request.handler;

import java.util.HashMap;
import java.util.Map;

import bryntum.crud.exception.CrudException;
import bryntum.crud.request.SyncStoreRequestHandler;
import bryntum.scheduler.domain.Resource;
import bryntum.scheduler.dao.Scheduler;

public class ResourceSyncHandler extends SyncStoreRequestHandler<Resource> {

    private final Scheduler app;

    public ResourceSyncHandler(Scheduler app) {
        this.app = app;
    }

    @Override
    public Resource getRecord(Map<String, Object> changes) {
        Integer id = (Integer) changes.get("Id");
        if (id == null) return null;

        Resource record = app.getResource(id);
        if (record == null) return null;

        if (changes.containsKey("Name")) {
            record.setName((String) changes.get("Name"));
        }

        return record;
    }

    @Override
    public Map<String, Object> add(Resource resource) throws CrudException {
        Map<String, Object> result = new HashMap<String, Object>();
        app.saveResource(resource);
        return result;
    }

    @Override
    public Map<String, Object> update(Resource resource, Map<String, Object> changes) throws CrudException {
        Map<String, Object> result = new HashMap<String, Object>();
        app.saveResource(resource);
        return result;
    }

    @Override
    public Map<String, Object> remove(Resource resource) throws CrudException {
        Map<String, Object> result = new HashMap<String, Object>();
        app.removeResource(resource);
        return result;
    }
}

File: examples/PHP demo/php/Bryntum/Scheduler/ResourceSyncHandler.php
<?php

namespace Bryntum\Scheduler;

class ResourceSyncHandler extends \Bryntum\CRUD\SyncHandler
{

    private $scheduler;

    public function __construct(&$scheduler)
    {
        $this->scheduler = &$scheduler;
        $this->phantomIdMap = &$scheduler->phantomIdMap['resources'];
    }

    public function add(&$resource)
    {
        $this->scheduler->saveResource($resource);
        return array();
    }

    public function update(&$resource)
    {
        $this->scheduler->saveResource($resource);
        return array();
    }

    public function remove($resource)
    {
        $this->scheduler->removeResource($resource);
        return array();
    }

    protected function onRecordAdded($record, &$recordResponse, $phantomId)
    {
        parent::onRecordAdded($record, $recordResponse, $phantomId);

        // let's keep phantom Id to real Id mapping
        $this->phantomIdMap[$phantomId] = $recordResponse['Id'];

        return $recordResponse;
    }
}
